src/main/java/thaumicboots/api/ItemBoots.java

```diff
@@ -0,0 +1,275 @@
+    }
+
+    protected void setBootsData() {
+        runicCharge = 0;
+        visDiscount = 0;
+        runBonus = 0.165F;
+        jumpBonus = 0.0D;
+        tier = 0;
+        steadyBonus = false; // this is the toggle for the longrunningbonus.
+        negateFall = false; // certain boots don't have fall damage in base.
+        waterEffects = false; // certain boots aren't hindered by being in the water.
+        longrunningbonus = 0.0F; // this is only for the comet boots, though it could be used for other boots.
+        iconResPath = "thaumicboots:electricVoid_16x";
+        armorResPath = "thaumicboots:model/electricbootsVoidwalker.png";
+        unlocalisedName = "ItemElectricVoid";
+        rarity = EnumRarity.rare; // this is less a variable, and more an indicator
+    }
+
+    public double getJumpModifier() {
+        return jumpBonus;
+    }
+
+    @SideOnly(Side.CLIENT)
+    public static double changeJump(double prevJump) {
+        double newJump = prevJump + 0.25;
+        if (newJump > 1) {
+            newJump = 0;
+        }
+        return newJump;
+    }
+
+    public static double isJumpEnabled(final ItemStack stack) {
+        return stack.stackTagCompound.getDouble(TAG_MODE_JUMP);
+    }
+
+    public static void setModeJump(ItemStack stack, double state) {
+        stack.stackTagCompound.setDouble(TAG_MODE_JUMP, state);
+    }
+
+    public float getSpeedModifier() {
+        return runBonus;
+    }
+
+    @SideOnly(Side.CLIENT)
+    public static double changeSpeed(double prevSpeed) {
+        double newSpeed = prevSpeed + 0.25;
+        if (newSpeed > 1) {
+            newSpeed = 0;
+        }
+        return newSpeed;
+    }
+
+    public static double isSpeedEnabled(final ItemStack stack) {
+        return stack.stackTagCompound.getDouble(TAG_MODE_SPEED);
+    }
+
+    public static void setModeSpeed(ItemStack stack, double state) {
+        stack.stackTagCompound.setDouble(TAG_MODE_SPEED, state);
+    }
+
+    // TODO: the part not from interfaces
+
+    @SideOnly(Side.CLIENT) // this method is important, it's what initializes the texture
+    public IIcon func_77617_a(int par1) {
+        return this.icon;
+    }
+
+    @SideOnly(Side.CLIENT)
+
+    public void registerIcons(IIconRegister ir) {
+        this.icon = ir.registerIcon(iconResPath);
+    }
+
+    @Override
+    public void addInformation(ItemStack stack, EntityPlayer par2EntityPlayer, List list, boolean par4) {
+        if (visDiscount > 0) {
+            list.add(
+                    EnumChatFormatting.DARK_PURPLE + StatCollector.translateToLocal("tc.visdiscount")
+                            + ": "
+                            + visDiscount
+                            + "%");
+        }
+    }
+
+    @Override
+    public String getArmorTexture(final ItemStack stack, final Entity entity, final int slot, final String type) {
+        return armorResPath;
+    }
+
+    @Override
+    public EnumRarity getRarity(final ItemStack stack) {
```

Comment:
Useless override

Reply:
Correction: the right way is just returning `rarity` here

Reply:
Funnily enough, that's what I originally did, I'll try it again

Reply:
rarity has been glitchy in my experience, this just ensures it works 100% of the time. Bizarre, I know.